### ISSUE TITLE
New unread responses flag was stuck

### ISSUE THREAD

**Issue by viraptor** [lobsters/lobsters#1065]
Today I noticed a new top bar layout. Shortly after I got a response with notification - that went away after reading as expected.

Some time after I got another notification, but I've not no unread responses and no messages: (same username as on GitHub)

![Screenshot_20220305-113146](https://user-images.githubusercontent.com/188063/156859640-e6ba339b-155b-4b5a-8a92-941ff78fee5d.jpg)

**Comment by pushcx** [comment]
I got one other report of this earlier today, too. Do you know if you had a reply or a private message? (The other reporter didn't recall.)

This feature was implemented in https://github.com/lobsters/lobsters/commit/e9b2f7466c8920731c7a51c9ab00bac866300a73 and https://github.com/lobsters/lobsters/commit/4d7d49b64c2884204eb1d64c565bb40691880ddf

**Comment by viraptor** [comment]
Originally it was a comment reply.

I did go to the private message list later (which was empty) - but that seems to have cleared the indicator anyway. No private messages received in months, so it wouldn't be the initial trigger.

**Comment by roryokane** [comment]
I, too, saw an indicator for one unread message, about three hours ago. When I tapped the link to https://lobste.rs/inbox, there were no unread replies, but the “1” indicator remained. I browsed to various pages, including Unread, All, Comments, Stories, and Messages, but the unread indicator persisted. I opened the home page and one or two individual story comment pages, and the inbox again. When I opened one of those pages, I forget which, I noticed that the unread indicator had been cleared. The unread indicator had been present for a total of 10 to 15 minutes.

Unlike the other two people who reported this, I have not received any valid notifications recently. My last story, comment, or message reply was a comment reply 23 days ago, and I had already read it weeks ago.

**Comment by pushcx** [comment]
That's really disconcerting, thank you. I'm going to disable the cache until I figure out what's wrong here.

**Comment by pushcx** [comment]
I went to look at the code and it popped out at me that I typo'd the string interpolation... then copied it everywhere I used the cache key, because I copy-and-paste things like that to avoid typos. Argh!

I'm going to leave this open a few days, though. It's certainly possible there's more bugs clearing the cache.

**Comment by azdle** [comment]
![Screenshot_20220319-083354](https://user-images.githubusercontent.com/262626/159123287-e7d1d3c8-96ff-4a4c-adf5-0934ea8f56b7.png)

Might be another one. I'm also seeing a notification count that popped up for the first time this morning, but I see 100. I have no unread messages and haven't posted anything lately, but also haven't clicked over to any of the other tabs in case keeping it there makes this easier to debug.

**Comment by kconner** [comment]
As of today I also see a red "100" and have no unread replies or messages.

**Comment by kconner** [comment]
This may be happening elsewhere: On the public user profile page, [including my own](https://lobste.rs/u/kevinc), comments posted always says 100. Stories submitted always says zero. I see the same values on every user's profile I try.

**Comment by pushcx** [comment]
Thanks for the heads up. Looks like e334ae41b45bf0a1499dae2deb97421370495fb6 broke something in the interface tweaks to the caching layer. I've reverted it for the moment.

**Comment by pushcx** [comment]
Yep, It certainly did. The fix is in ef642dff691c6e137c96da3200c719d9e774d7e8. This was a really dumb bug with unfortunate side effects. It started emailing everyone using mailing list mode basically every story or comment back to the creation of the site. I messaged an apology to all users who got spammed.

**Comment by RaymiiOrg** [comment]
I might have it the other way around, new comment reply's don't trigger the red number next to the notification icon. If I open the link, I do have unread post replies. This started, I think, after receiving a private message.
